Create decentralizing-xmtp.md (#800)
xmtp/xmtp-dot-org: src/pages/decentralizing-xmtp.md

# Decentralizing XMTP

> Today we're announcing a new chapter for XMTP: a decentralized messaging network powered by the MLS Standard - the most secure and private messaging protocol available to developers.  Our entire protocol is open source, from the [node software](https://github.com/xmtp/xmtp-node-go) to [the SDKs](https://github.com/xmtp) to the [Converse app](https://github.com/ephemerahq/converse-app).

Reviewer: ```suggestion
Today we're announcing a new chapter for XMTP: a decentralized messaging network powered by the [Messaging Layer Security standard](https://messaginglayersecurity.rocks/) standard - the most secure and private messaging protocol available to developers.  Our entire protocol is open source, from the [node software](https://github.com/xmtp/xmtp-node-go) to the [SDKs](https://github.com/xmtp) to the [Converse app](https://github.com/ephemerahq/converse-app).
```

> Until now, the backend of the network has been centralized. All the nodes are run by Ephemera. It’s time to change that.

Reviewer: ```suggestion
Until now, the backend of the network has been centralized. All the nodes are run by [Ephemera](https://ephemerahq.com/). It’s time to change that.
```

## Why decentralization matters

1. Devs are tired of getting rugged by platforms who can change the rules any time it suits their business interests.
2. Devs want to be owners of the network and share in the economic value that they are creating

> 3. Devs should feel confident that the platform they are building on top of will not live or die with a single company.

Reviewer: ```suggestion
3. Devs should feel confident that the platform they are building on will not live or die with a single company.
```

## How does the network work?

> In this decentralized network there are three types of actor. Each plays a distinct role in delivering a reliable, high-performance, end-to-end encrypted messaging service for users.

Reviewer: ```suggestion
In this decentralized network there are three types of actors. Each plays a distinct role in delivering a reliable, high-performance, end-to-end encrypted messaging service for users.
```

3. **Payers** will fairly compensate node operators for the cost of delivering and storing messages.

Two distributed systems will coordinate these interactions.

1. The **XMTP broadcast network:** a small, permissioned set of node operators replicating messages across diverse organizations and geographies to ensure availability.
2. The **XMTP appchain:** an L3 blockchain securing all metadata that require strict ordering. Four smart contracts will manage this data:
    - The `Nodes` contract manages the list of broadcast network nodes, ensuring that only authorized nodes participate in the network.
    - The `Payers` contract manages the payment of messaging fees to node operators, allowing applications and organizations to cover network costs without requiring users to pay directly.
    - The `IdentityUpdates` contract manages the list of wallet addresses associated with each XMTP inbox, enabling users to use multiple authorized wallet addresses to send and receive messages from a single inbox.
    - The `GroupMessages` contract manages group membership changes, ensuring that all clients have an identical view of group additions and removals in the correct order.

> For more specifics, follow the team's progress in [the xmtpd repo](https://github.com/xmtp/xmtpd/issues/118). A detailed XIP outlining the decentralization process will be published shortly.

Reviewer: ```suggestion
For more specifics, follow the team's progress in the [xmtpd repo](https://github.com/xmtp/xmtpd/issues/118). A detailed XIP outlining the decentralization process will be published shortly.
```

![Diagram of actors in the XMTP decentralized network](../../static/img/decentralizing-xmtp-1.png)

### Timeline and key milestones

We expect to launch a testnet of the new decentralized network at the end of November, 2024. The testnet will be unincentivized with no fees charged to payers and no rewards distributed to node operators.

We expect to add incentives to the testnet in Q1 2025 and to be ready for final audits and mainnet in H2 2025.

## FAQ

**1. I have an inbox app powered by XMTP. What does this mean for me?**

Most changes to support the new decentralized network will happen under the hood and not require any changes to your application code beyond updating to a newer SDK version and providing a payer service URL or signer.

Today, the XMTP network is free to use. The decentralized network will have a cost associated with each message sent. We are designing the network architecture to minimize these costs, with the goal of making them practical for application developers to sponsor on behalf of their users. Mainnet cost estimates will be published in February 2025 after completing load testing of the testnet and fee mechanisms.

> To sponsor transactions on behalf of your users client applications will need to pass in the URL to a payer service as part of client creation. The payer service is a small, stateless, backend service that holds a set of keys used to sponsor transactions and signs messages using the application’s account. While this may sound similar to a smart wallet paymaster, the payer service serves additional crucial functions: it enables off-chain bookkeeping and enhances user privacy by preventing the linking of senders to their messages. The service can be self hosted by developers and will be responsible for deciding which messages should be subsidized by the application.

Reviewer: ```suggestion
To sponsor transactions on behalf of your users, client applications will need to pass in the URL to a payer service as part of client creation. The payer service is a small, stateless, backend service that holds a set of keys used to sponsor transactions and signs messages using the application’s account. While this may sound similar to a smart wallet paymaster, the payer service serves additional crucial functions: it enables off-chain bookkeeping and enhances user privacy by preventing the linking of senders to their messages. The service can be self hosted by developers and will be responsible for deciding which messages should be subsidized by the application.
```

> If you are interested in building a “payer-as-a-service” platform that can be used by many applications you should [reach out to us](mailto://hi@xmtp.org). We’d love to help support the development of a platform that helps app developers avoid running their own payer backend service.

Reviewer: ```suggestion
If you are interested in building a “payer-as-a-service” platform that can be used by many applications you should [reach out to us](mailto:hi@xmtp.org). We’d love to help support the development of a platform that helps app developers avoid running their own payer backend service.
```

Complete guides for setting up a payer service and funding your account for sponsoring messages will be provided at least 3 months before the launch of mainnet.

> **2. I have a bot or broadcast app powered by XMTP? What does this mean for me?**

Reviewer: ```suggestion
**2. I have a bot or broadcast app powered by XMTP. What does this mean for me?**
```

**3. What is the XMTP appchain?**

The XMTP appchain is an L3 blockchain built as an Arbitrum Orbit rollup that settles onto Base, an Optimism L2 that has traction both with developers and consumers. It handles critical metadata that needs strict ordering with higher latency tolerance. This setup offers a balance of scalability, security, and cost-effectiveness, making it ideal for coordinating clients, node operators, and payers.

**4. What is the XMTP broadcast network?**

![Diagram of XMTP broadcast network](../../static/img/decentralizing-xmtp-2.png)

The XMTP broadcast network is a small group of 5–20 nodes, each run by a separate organization. It replicates messages across diverse geographies to ensure availability, control latency, and enhance performance using scalable infrastructure.

> **5. Why can’t anyone run a node?**

Reviewer: ```suggestion
**5. Why can’t just anyone run a node?**
```

> It’s helpful to go back to our design goals for decentralization.

Reviewer: ```suggestion
It’s helpful to go back to the design goals for the decentralization of XMTP.
```

1. Have a network with the performance and latency of web2 messaging services
2. Have extremely low messaging costs
3. Be credibly neutral and censorship resistant

> A smaller network of nodes is better for achieving these goals. For a message to be distributed to all node operators requires less bandwidth and fewer hops with a smaller network, and there are “less mouths to feed” to reach economic sustainability for node operators. This allows us to deliver high performance at a relatively low cost.

Reviewer: ```suggestion
A smaller network of nodes is better for achieving these goals. For a message to be distributed to all node operators requires less bandwidth and fewer hops with a smaller network, and there are “less mouths to feed” to reach economic sustainability for node operators. This allows XMTP to deliver high performance at a relatively low cost.
```

A network of 20 node operators can still be credibly neutral and censorship resistant so long as the group of operators is sufficiently diverse to prevent collusion between operators. And the network is designed to achieve its censorship resistance so long as there is a single honest node.

This architecture reflects the blockchain trilemma of balancing security, scalability, and decentralization. Security is non-negotiable for a messaging protocol, and we believe this design strikes the right balance between scalability and decentralization for our use case. Messaging demands higher throughput and lower latency than financial applications, while having different trust requirements.

> The security of the network extends beyond the node operators themselves. Just as optimistic rollups use fault proofs, our network allows any participant to submit cryptographically verifiable `MisbehaviorReports` when nodes violate protocol rules. Looking ahead, we are developing an automated node selection process (likely involving staking mechanisms) to ensure long-term sustainability and decentralization.

Reviewer: ```suggestion
The security of the network extends beyond the node operators themselves. Just as optimistic rollups use fault proofs, the XMTP network allows any participant to submit cryptographically verifiable `MisbehaviorReports` when nodes violate protocol rules. Looking ahead, we are developing an automated node selection process (likely involving staking mechanisms) to ensure long-term sustainability and decentralization.
```
